### DISCUSSION TITLE
Question: Node.js Native Addons

### DISCUSSION THREAD

**Discussion by ghost** [denoland/deno#15421] [Q&A] [answered]
Hello, had a question regarding compatibility with Node.js addons. Was reading about this years ago. I've seen some compatibility articles and issues regarding the subject.  I can see that deno_std is not currently 100% stable. It is possible to load a Node.js addon written in TypeScript that utilities Node.js C/C++ addons either via CommonJS or the URL caching? 

https://github.com/denoland/deno_std/blob/main/node/README.md
https://github.com/denoland/deno/issues/2644

**Comment by bartlomieju** [comment, marked as answer]
No, it's currently not possible We have a PR going to add support for it in #13633

**Comment by ghost** [comment]
Excellent. Thank you.

**Reply by andykais** [reply]
just as a callout, the current work will add NAPI (Node-API) support, but not NAN (Native Abstractions for Node.js) https://github.com/denoland/deno/issues/13253. So not every native node library will be compatible

**Reply by ghost** [reply]
Good to know. I am relatively new to contributing and discussing here. Perhaps the maintainers could let us know if there are plans to introduce the NAN interface at some point.

The addon that I am writing my spare time I wanted to be compatible with Deno and Node v16.16.0 over here at https://github.com/mtmulch1/TypeScriptSecurityAddons

**Comment by ghost** [comment]
Just glancing at some of the commits associated with this pull request it seems there has been much success porting over the direct V8 methods on simpler methods and excuse my Rust terminology and creating an interface to bind to the cc file containing the V8 method?

**Reply by littledivy** [reply]
There is no interface to bind with a c++ file. Node-API (previously NAPI) defines a stable ABI which is not tied to just C++. The PR implements most of this ABI, the goal is too run _all_ NAPI-compatible modules not just C++ ones.

**Reply by ghost** [reply]
One of the commits that I was referring to was this one https://github.com/denoland/rusty_v8/pull/880/commits/49c65676bd4894389820275d7a7dfc7e0c6d1ad9 when you are returning a V8 isolate are you still providing the independent copy of the runtime and allowing Rust to handle the garbage collection defined by the byte size?
